### PR TITLE
Migrate pkg/kubelet/nodestatus to structured logging

### PR DESCRIPTION
**Pull request by QiWang19**
Migrate pkg/kubelet/nodestatus to structured logging

#### What type of PR is this?

/kind cleanup
#### What this PR does / why we need it:
Migrate pkg/kubelet/nodestatus to structured logging
#### Which issue(s) this PR fixes:
Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
Migrate pkg/kubelet/nodestatus to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
/ok-to-test

please change:
> Migrate pkg/kubelet/nodestatus to structured logging

to:

> NONE

**Comment by QiWang19** [comment]
> 
> ~please change:~
> ~> Migrate pkg/kubelet/nodestatus to structured logging~
> 
> ~> NONE~


Updated.

File: pkg/kubelet/nodestatus/setters.go [hunk 1]
```diff
@@ -191,11 +191,11 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 
 				if existingHostnameAddress == nil {
 					// no existing Hostname address found, add it
-					klog.Warningf("adding overridden hostname of %v to cloudprovider-reported addresses", hostname)
+					klog.InfoS("adding overridden hostname to cloudprovider-reported addresses", "hostname", hostname)
 					nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeHostName, Address: hostname})
 				} else if existingHostnameAddress.Address != hostname {
 					// override the Hostname address reported by the cloud provider
-					klog.Warningf("replacing cloudprovider-reported hostname of %v with overridden hostname of %v", existingHostnameAddress.Address, hostname)
+					klog.InfoS("replacing cloudprovider-reported hostname with overridden hostname", "cloudprovider-reportedHostname", existingHostnameAddress.Address, "overriddenHostname", hostname)
```
Comment (ehashman) [inline pkg/kubelet/nodestatus/setters.go:198]: I wonder if we should pick different labels for these... @serathius for replacements in general do we have any guidance on old/new? Might make this a little easier to grok.

cc @dims do you have an opinion?

**Review by ehashman — COMMENTED** [review]
/priority important-soon
/triage accepted

File: pkg/kubelet/nodestatus/setters.go (hunk 1 above)
Comment (dims) [reply pkg/kubelet/nodestatus/setters.go:198]: +1 to make it simpler ... say `cloudprovider-hostname` and `override-hostname`

Comment (serathius) [reply pkg/kubelet/nodestatus/setters.go:198]: There is no strong guidance on prefixes based on context. Some cases like this is are unique enought that we just want to make sure that they make sense in context of this specific log. Of cause if it make sense in context, simpler prefix is better like `old`/`new` > `override`/`cloudprovider`. Don't know this code enough if `old`/`new` makes sense here.

+1 for dims suggestion with simplified prefix, but let's use lower camel case as in migration guide. So
`cloudproviderHostname` and `overrideHostname`

Comment (dims) [reply pkg/kubelet/nodestatus/setters.go:198]: +1 from me!

File: pkg/kubelet/nodestatus/setters.go [hunk 2]
```diff
@@ -574,7 +574,7 @@ func ReadyCondition(
 				recordEventFunc(v1.EventTypeNormal, events.NodeReady)
 			} else {
 				recordEventFunc(v1.EventTypeNormal, events.NodeNotReady)
-				klog.Infof("Node became not ready: %+v", newNodeReadyCondition)
+				klog.InfoS("Node became not ready", "node", klog.KObj(node), "notReadyCondition", newNodeReadyCondition)
```
Comment (serathius) [inline pkg/kubelet/nodestatus/setters.go:577]:
```suggestion
				klog.InfoS("Node became not ready", "node", klog.KObj(node), "condition", newNodeReadyCondition)
```

**Comment by serathius** [comment]
Just one comment, otherwise LGTM

File: pkg/kubelet/nodestatus/setters.go (hunk 2 above)
Comment (QiWang19) [reply pkg/kubelet/nodestatus/setters.go]: fixed

File: pkg/kubelet/nodestatus/setters.go [hunk 3]
```diff
@@ -340,13 +340,13 @@ func MachineInfo(nodeName string,
...
 				if old, ok := node.Status.Capacity[k]; !ok || old.Value() != v.Value() {
-					klog.V(2).Infof("Update capacity for %s to %d", k, v.Value())
+					klog.V(2).InfoS("Updated capacity", "devicePlugin", k, "capacity", v.Value())
 				}
 				node.Status.Capacity[k] = v
 			}
 
 			for _, removedResource := range removedDevicePlugins {
-				klog.V(2).Infof("Set capacity for %s to 0 on device removal", removedResource)
+				klog.V(2).InfoS("Set capacity for removed resource to 0 on device removal", "removedDevice", removedResource)
```
Comment (serathius) [inline pkg/kubelet/nodestatus/setters.go:349]:
```suggestion
				klog.V(2).InfoS("Set capacity for removed resource to 0 on device removal", "device", removedResource)
```

Comment (serathius) [reply pkg/kubelet/nodestatus/setters.go:349]: Missed that one

**Comment by serathius** [comment]
Overall looks good, please update the release notes

**Comment by QiWang19** [comment]
/assign @mrunalp
Needs approved label.

**Pull request merged**